homeassistant/components/light/lw12wifi.py

```diff
@@ -0,0 +1,210 @@
+"""
+Support for Lagute LW-12 WiFi LED Controller.
+
+Configuration example:
+
+light:
+  - platform: lw12wifi
+    # Host name or IP of LW-12 LED stripe to control
+    host: 192.168.0.1
+    # (Optional) Some firmware versions of the LW-12 controller listen
+    # on different ports.
+    port: 5000
+    # (Optional) Friendly name
+    name: LW-12 FC
+    # (Optional) Effect to use as default (Default=None).
+    # For a full list of supported effects see:
+    # https://github.com/jaypikay/python-lw12/blob/v0.9.2/lw12.py#L54
+    effect: "Gradient Cyan"
+    # (Optional) Transition speed for effects (Value: 0-255)
+    transition: 10
+    # (Optional) LED Brightness (Value: 0-255)
+    brightness: 10
+    # (Optional) RGB value array as standard color to set.
+    # This option is is ignored when effect is set.
+    rgb: [255, 255, 255]
+"""
+
+import asyncio
+from enum import Enum
+import logging
+import socket
+import time
+
+import voluptuous as vol
+
+from homeassistant.components.light import (
+    ATTR_BRIGHTNESS, ATTR_EFFECT, ATTR_RGB_COLOR, ATTR_TRANSITION, DOMAIN,
+    Light, PLATFORM_SCHEMA, SUPPORT_BRIGHTNESS, SUPPORT_EFFECT,
+    SUPPORT_RGB_COLOR, SUPPORT_TRANSITION
+)
+from homeassistant.const import (
+    CONF_BRIGHTNESS, CONF_EFFECT, CONF_HOST, CONF_NAME, CONF_PORT, CONF_RGB
+)
+import homeassistant.helpers.config_validation as cv
+
+
+REQUIREMENTS = ['lw12==0.9.2']
+
+_LOGGER = logging.getLogger(__name__)
+
+CONF_TRANSITION = 'transition'
+
+DEFAULT_NAME = 'LW-12 FC'
+DEFAULT_PORT = 5000
+DEFAULT_RGB = [255, 255, 255]
+DEFAULT_BRIGHTNESS = 255
+DEFAULT_EFFECT = None
+DEFAULT_TRANSITION = 128
+
+DOMAIN = 'lw12wifi'
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_HOST): cv.string,
+    vol.Optional(CONF_PORT, default=DEFAULT_PORT): cv.port,
+    vol.Optional(CONF_RGB, default=DEFAULT_RGB):
+    vol.All(list, vol.Length(min=3, max=3),
```

Comment:
The indentation makes this quite hard to read, could you maybe consider indenting the `vol.All` blocks?